src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	public final String value;
+    protected boolean isPrivate;
+    
+    
+    public Contact(){
```

Comment:
As mentioned during the tutorial, it is entirely possible to push the current subclass constructor code inside the parent class, including the validation methods and regex